be/src/vec/exec/vrepeat_node.cpp

```diff
@@ -0,0 +1,244 @@
+    ss << "The output slots size " << _output_slots.size() 
+        << " is not equal to the sum of child_slots_size " << _child_slots.size()
+        << ",virtual_slots_size " << _virtual_tuple_desc->slots().size();
+    if (_output_slots.size() != (_child_slots.size() + _virtual_tuple_desc->slots().size())) {
+        return Status::InternalError(ss.str());
+    }
+
+    return Status::OK();
+}
+
+Status VRepeatNode::open(RuntimeState* state) {
+    VLOG_CRITICAL << "VRepeatNode::open";
+    SCOPED_TIMER(_runtime_profile->total_time_counter());
+    RETURN_IF_ERROR(RepeatNode::open(state));
+    return Status::OK();
+}
+
+Status VRepeatNode::get_repeated_block(Block* child_block, int repeat_id_idx, Block* output_Block) {
+    VLOG_CRITICAL << "VRepeatNode::get_repeated_block";
+    DCHECK(child_block != nullptr);
+    DCHECK_EQ(output_Block->rows(), 0);
+
+    size_t child_column_size = child_block->columns();
+    size_t column_size = _output_slots.size();
+    bool mem_reuse = output_Block->mem_reuse();
+    DCHECK_EQ(child_column_size, _child_slots.size());
+    DCHECK_LE(child_column_size, column_size);
+    std::vector<vectorized::MutableColumnPtr> columns(column_size);
+    for (size_t i = 0; i < column_size; i++) {
+        if (mem_reuse) {
+            columns[i] = std::move(*output_Block->get_by_position(i).column).mutate();
+        } else {
+            columns[i] = _output_slots[i]->get_empty_mutable_column();
+        }
+    }
+
+    /* Fill all slots according to child, for example:select tc1,tc2,sum(tc3) from t1 group by grouping sets((tc1),(tc2));
+     * insert into t1 values(1,2,1),(1,3,1),(2,1,1),(3,1,1);
+     * slot_id_set_list=[[0],[1]],repeat_id_idx=0,
+     * child_block 1,2,1 | 1,3,1 | 2,1,1 | 3,1,1
+     * output_block 1,null,1,1 | 1,null,1,1 | 2,nul,1,1 | 3,null,1,1
+     */
+    size_t cur_col = 0;
+    for (size_t i = 0; i < child_column_size; i++) {
+        const ColumnWithTypeAndName& src_column = child_block->get_by_position(i);
+
+        DCHECK_EQ(_child_slots[i]->type().type, _output_slots[cur_col]->type().type);
+        DCHECK_EQ(_child_slots[i]->col_name(), _output_slots[cur_col]->col_name());
+
+        std::set<SlotId>& repeat_ids = _slot_id_set_list[repeat_id_idx];
+        bool is_repeat_slot = _all_slot_ids.find(_output_slots[cur_col]->id()) != _all_slot_ids.end();
+        bool is_set_null_slot = repeat_ids.find(_output_slots[cur_col]->id()) == repeat_ids.end();
+
+        if (is_repeat_slot) {
+            DCHECK(_output_slots[cur_col]->is_nullable());
+            auto* nullable_column = reinterpret_cast<ColumnNullable *>(columns[cur_col].get());
+            auto& null_map = nullable_column->get_null_map_data();
+            auto* column_ptr = columns[cur_col].get();
+
+            // set slot null not in repeat_ids
+            if (is_set_null_slot) {
+                for (size_t j = 0; j < src_column.column->size(); ++j) {
+                    nullable_column->insert_data(nullptr, 0);
```

Comment:
This code maybe can not SIMD? maybe should call resize function to set nullable_column size to SIMD